src/lib/pipeline_configuration.py

```diff
@@ -0,0 +1,891 @@
+    Use for columns containing yes/no/ambivalent codes which can vary between message objects for the same person.
+    """
+
+    CONCATENATE = "CONCATENATE"
+    """
+    Join all the values in the input message objects, separating them by a ';'. For example, the messages "a", "b", and
+    "c" will be folded to "a;b;c". Uses for columns which differ and where all the original messages should be 
+    preserved.
+    """
+
+    MATRIX = "MATRIX"
+    """
+    Set the folded value to Codes.MATRIX_1 if either input is Codes.MATRIX_1, otherwise to Codes.MATRIX_0 or the 
+    appropriate control code. Use for columns in matrix representation.
+    """
+
+
+class CodingConfiguration(object):
+    def __init__(self, coding_mode, code_scheme, coded_field, folding_mode, analysis_file_key=None, cleaner=None):
+        assert coding_mode in {CodingModes.SINGLE, CodingModes.MULTIPLE}
+
+        self.coding_mode = coding_mode
+        self.code_scheme = code_scheme
+        self.coded_field = coded_field
+        self.analysis_file_key = analysis_file_key
+        self.folding_mode = folding_mode
+        self.cleaner = cleaner
+
+
+# TODO: Rename CodingPlan to something like DatasetConfiguration?
+class CodingPlan(object):
+    def __init__(self, raw_field, coding_configurations, raw_field_folding_mode, coda_filename=None, ws_code=None,
+                 time_field=None, run_id_field=None, icr_filename=None, id_field=None, code_imputation_function=None):
+        self.raw_field = raw_field
+        self.time_field = time_field
+        self.run_id_field = run_id_field
+        self.coda_filename = coda_filename
+        self.icr_filename = icr_filename
+        self.coding_configurations = coding_configurations
+        self.code_imputation_function = code_imputation_function
+        self.ws_code = ws_code
+        self.raw_field_folding_mode = raw_field_folding_mode
+
+        if id_field is None:
+            id_field = "{}_id".format(self.raw_field)
+        self.id_field = id_field
+
+
+class PipelineConfiguration(object):
+
+    RQA_CODING_PLANS = None
+
+    DADAAB_RQA_CODING_PLANS = [
+        CodingPlan(raw_field="rqa_s01e01_raw",
+                   time_field="sent_on",
+                   run_id_field="rqa_s01e01_run_id",
+                   coda_filename="s01e01.json",
+                   icr_filename="s01e01.csv",
+                   coding_configurations=[
+                       CodingConfiguration(
+                           coding_mode=CodingModes.MULTIPLE,
+                           code_scheme=CodeSchemes.DADAAB_S01E01_REASONS,
+                           coded_field="rqa_s01e01_coded",
+                           analysis_file_key="rqa_s01e01_",
+                           folding_mode=FoldingModes.MATRIX
+                       )
+                   ],
+                   ws_code=CodeSchemes.WS_CORRECT_DATASET.get_code_with_match_value("dadaab s01e01"),
+                   raw_field_folding_mode=FoldingModes.CONCATENATE),
```

Comment:
```suggestion
                   raw_field_folding_mode=FoldingModes.CONCATENATE),
                   
```
(makes things a bit easier to read)